cmd/influxd/upgrade/upgrade.go

```diff
@@ -40,12 +41,36 @@ type configV1 struct {
 		Dir    string `toml:"dir"`
 		WALDir string `toml:"wal-dir"`
 	} `toml:"data"`
+	Http struct {
```

Comment:
please don't build the url using string concatination.  Please use the `net.URL` object.